src/config/project.rs

```diff
@@ -180,17 +180,26 @@ pub struct ProjectConfig {
 }
 
 impl ProjectConfig {
-    fn parse(dir: &Utf8Path, metadata: &serde_json::Value) -> Result<Self> {
+    fn parse(dir: &Utf8Path, metadata: &serde_json::Value, cargo_metadata: &Metadata) -> Result<Self> {
         let mut conf: ProjectConfig = serde_json::from_value(metadata.clone())?;
         conf.config_dir = dir.to_path_buf();
         let dotenvs = load_dotenvs(dir)?;
         overlay_env(&mut conf, dotenvs)?;
-        if conf.site_root == "/" || conf.site_root == "." {
+        if conf.site_root == "/" || conf.site_root == "." || conf.site_root == "CARGO_TARGET_DIR" {
```

Comment:
Should this include the synonym `CARGO_BUILD_TARGET_DIR` as well? https://doc.rust-lang.org/cargo/reference/config.html#buildtarget-dir